### PR TITLE
4170 documents

### PR DESCRIPTION
**Pull request by briaguya**
# Description
update importer to support importing documents and official document pages
https://github.com/cityofaustin/techstack/issues/4170

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 
at the moment circleCI is having issues, but if you run it locally you can import an official document page and watch the magic happen!

I imported "Official complaint and discipline documents" from staging,
http://joplin-pr-4170-documents.herokuapp.com/admin/pages/11/edit/ 

https://joplin-austin-gov-static.s3.amazonaws.com/review/4170-documents/media/documents/2019-0617.pdf

# Checklist:
- [x] Request reviewers
- [ ] Slack-out message for review
- [ ] Link this PR in the issue
- [ ] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/api/schema.py [hunk 1]
```diff
@@ -22,7 +22,7 @@
 from pages.service_page.models import ServicePage
 from pages.information_page.models import InformationPage
 from pages.department_page.models import DepartmentPage, DepartmentPageDirector, DepartmentPageTopPage, DepartmentPageRelatedPage
-from pages.official_documents_page.models import OfficialDocumentPage, OfficialDocumentPageOfficialDocument
+from pages.official_documents_page.models import OfficialDocumentPage, OfficialDocumentPageDocument
```
Comment (Niicck) [inline joplin/api/schema.py:25]: good call

File: joplin/base/management/commands/load_joplin_data.py [hunk 2]
```diff
@@ -61,20 +61,16 @@ def load_fixture(operation_name, fixture_name, condition):
...
                     print("Adding prod datadump")
                     run_load_data_command('db/system-generated/prod.datadump.json')
                     DeploymentLog(operation="load_data", value="prod", completed=True).save()
-                elif (LOAD_DATA == "staging"):
-                    print("Adding staging datadump")
-                    run_load_data_command('db/system-generated/staging.datadump.json')
-                    DeploymentLog(operation="load_data", value="staging", completed=True).save()
-                elif (LOAD_DATA == "dummy"):
-                    print("Adding dummy datadump")
+                elif LOAD_DATA == 'fixtures':
```
Comment (Niicck) [inline joplin/base/management/commands/load_joplin_data.py:68]: 👍

File: joplin/pages/official_documents_page/factories.py [hunk 3]
```diff
@@ -1,15 +1,164 @@
...
+            # A list of topics were passed in, use them
+            for official_documents_page_document in extracted['official_documents_page_documents']:
+                OfficialDocumentPageDocumentFactory.create(page=self, **official_documents_page_document)
+            return
+
+
+class DocumentFactory(factory.DjangoModelFactory):
+    @classmethod
+    def create(cls, *args, **kwargs):
+        # todo: document file stuff here
```
Comment (Niicck) [inline joplin/pages/official_documents_page/factories.py:32]: what kind of document file stuff?

File: joplin/pages/official_documents_page/factories.py [hunk 4]
```diff
@@ -1,15 +1,164 @@
...
+    # Check if a document with the same hash has already been imported
+    try:
+        document = Document.objects.get(file_hash=file_hash)
+    except Document.DoesNotExist:
+        document = None
+    if document:
+        return document
+
+    # It has not been imported, let's do it!
+    document = DocumentFactory.create(file=ContentFile(response.content, name=file_name), title=file_name)
```
Comment (Niicck) [inline joplin/pages/official_documents_page/factories.py:163]: I appreciate how long it took to make this work.

File: joplin/settings.py [hunk 5]
```diff
@@ -403,7 +403,7 @@ def show_toolbar(request):
         # All non-production apps share a staging/media folder
         AWS_LOCATION = f"review/{os.getenv('CIRCLE_BRANCH')}/static"
         AWS_IS_GZIPPED = True
-        MEDIAFILES_LOCATION = 'staging/media'
+        MEDIAFILES_LOCATION = f"review/{os.getenv('CIRCLE_BRANCH')}/media"
```
Comment (Niicck) [inline joplin/settings.py:406]: YES

File: joplin/pages/official_documents_page/factories.py (hunk 3 above)
Comment (briaguya) [reply joplin/pages/official_documents_page/factories.py:32]: whoops

File: joplin/pages/official_documents_page/factories.py (hunk 4 above)
Comment (easherma) [reply joplin/pages/official_documents_page/factories.py:162]: FYI wagtail-factories has a document factory:
https://github.com/mvantellingen/wagtail-factories/blob/819666151f8c83317bdd0019074173ba2ade85eb/src/wagtail_factories/factories.py#L142

Which leverages factory boy's Django FileField:
https://factoryboy.readthedocs.io/en/latest/orms.html#extra-fields

**Pull request merged**